crates/ty_ide/src/goto.rs

```diff
@@ -828,6 +867,495 @@ f(**kwargs<CURSOR>)
+         --> main.py:7:19
+          |
+        5 |             else:
+        6 |                 ab = 1
+        7 |             print(ab)
+          |                   ^^
+          |
+
+        info[goto-type-definition]: Type definition
+         --> main.py:6:17
+          |
+        4 |                 ab = 2
+        5 |             else:
+        6 |                 ab = 1
+          |                 ^^
+        7 |             print(ab)
+          |
+        info: Source
+         --> main.py:7:19
+          |
+        5 |             else:
+        6 |                 ab = 1
+        7 |             print(ab)
+          |                   ^^
+          |
+        ");
+    }
+
+    #[test]
+    fn goto_def_local_load_bind_decl() {
+        let test = cursor_test(
+            r#"
+            ab: int
+            ab = 1
+            print(a<CURSOR>b)
+        "#,
+        );
+
+        assert_snapshot!(test.goto_definition(), @r"
+        info[goto-type-definition]: Type definition
+         --> main.py:3:13
+          |
+        2 |             ab: int
+        3 |             ab = 1
+          |             ^^
+        4 |             print(ab)
+          |
+        info: Source
+         --> main.py:4:19
+          |
+        2 |             ab: int
+        3 |             ab = 1
+        4 |             print(ab)
+          |                   ^^
+          |
+        ");
+    }
+
+    #[test]
+    fn goto_def_local_first_store() {
+        let test = cursor_test(
+            r#"
+            a<CURSOR>b = 1
+            print(ab)
+            ab = 2
+        "#,
+        );
+
+        assert_snapshot!(test.goto_definition(), @"No goto target found");
+    }
+
+    #[test]
+    fn goto_def_local_second_store() {
+        let test = cursor_test(
+            r#"
+            ab = 1
+            print(ab)
+            a<CURSOR>b = 2
+        "#,
+        );
+
+        assert_snapshot!(test.goto_definition(), @"No goto target found");
+    }
+
+    #[test]
+    fn goto_def_local_loadstore() {
+        let test = cursor_test(
+            r#"
+            ab = 1
+            print(ab)
+            a<CURSOR>b += 2
+            print(ab)
+        "#,
+        );
+
+        assert_snapshot!(test.goto_definition(), @"No goto target found");
+    }
+
+    #[test]
+    fn goto_def_class() {
+        let test = cursor_test(
+            r#"
+            class AB(val: int):
+                self.myval = val
+
+            x = A<CURSOR>B(5)
+        "#,
+        );
+
+        assert_snapshot!(test.goto_definition(), @r"
+        info[goto-type-definition]: Type definition
+         --> main.py:2:19
+          |
+        2 |             class AB(val: int):
+          |                   ^^
+        3 |                 self.myval = val
+          |
+        info: Source
+         --> main.py:5:17
+          |
+        3 |                 self.myval = val
+        4 |
+        5 |             x = AB(5)
+          |                 ^^
```

Comment:
Note that this is only working because the class is defined in the same scope. If `x = AB(5)` was in a function it would fail to resolve, because we don't look into parent scopes. yet.